internal/diff/unified.go

```diff
@@ -182,16 +182,8 @@ func (u Unified) Format(f fmt.State, r rune) {
 			}
 		}
 		fmt.Fprint(f, "@@")
-		if fromCount > 1 {
-			fmt.Fprintf(f, " -%d,%d", hunk.FromLine, fromCount)
-		} else {
-			fmt.Fprintf(f, " -%d", hunk.FromLine)
-		}
-		if toCount > 1 {
-			fmt.Fprintf(f, " +%d,%d", hunk.ToLine, toCount)
-		} else {
-			fmt.Fprintf(f, " +%d", hunk.ToLine)
-		}
+		fmt.Fprintf(f, " -%d,%d", fromCount, fromCount)
+		fmt.Fprintf(f, " +%d,%d", hunk.ToLine, toCount)
```

Comment:
This still doesn't quite align with how `git diff` generates patches in the special cases. Basically, it's like this:

New file: `@@ -0,0 +1,2 @@`
Removed file: `@@ -1,7 +0,0 @@`

And then you have normal hunks, which are handled normally.

I think we'd need to do this to get the exact same semantics:

```suggestion
		if fromCount == 0 {
			fmt.Fprintf(f, " -0,0 +%d,%d", hunk.ToLine, toCount)
		} else if toCount == 0 {
			fmt.Fprintf(f, " -%d,%d +0,0", hunk.FromLine, fromCount)
		} else {
			fmt.Fprintf(f, " -%d,%d +%d,%d", hunk.FromLine, fromCount, hunk.ToLine, toCount)
		}
```